logictest: fix comparison logic

otan opened the pull request:
`9d2c054fd3e9e8c6a7d73f4e464ebd32ee82b8a2` introduced a bug in the
commit logic where it would only check logic test errors for the first
line, since it only compares the first column of results. This is now
fixed to compare every row.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55876)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55876#-:-MKHLOtZ5BT3FdEcnC64:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/logictest/logic.go, line 2741 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55876#-MKHL8mH-gfCDbIx2eWp:-MKHL8mH-gfCDbIx2eWq:b8bhsjl) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e5eb0cc62cbcf2f10005e538997bca76f074b10/pkg/sql/logictest/logic.go#L2741)):*
> ```Go
> 				if colT == 'F' {
> 					var err error
> 					resultMatches, err = floatsMatch(expected, actual)
> ```

~Your first version was correct - we now don't check whether `resultMatches` is `true` in this case.~ Ok, nvm.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55876)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/logictest/logic.go, line 2741 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55876#-MKHL8mH-gfCDbIx2eWp:-MKHLc4wBiDjZ9700Fsh:bxs9oh) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e5eb0cc62cbcf2f10005e538997bca76f074b10/pkg/sql/logictest/logic.go#L2741)):*
<details><summary><i>Previously, yuzefovich wrote…</i></summary><blockquote>

~Your first version was correct - we now don't check whether `resultMatches` is `true` in this case.~ Ok, nvm.
</blockquote></details>

Oh wait, we still don't check `resultMatches` here.


<!-- Sent from Reviewable.io -->

otan commented:
___
*[pkg/sql/logictest/logic.go, line 2741 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55876#-MKHL8mH-gfCDbIx2eWp:-MKHLuZZ12G_z6Y8nykO:by0bezo) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e5eb0cc62cbcf2f10005e538997bca76f074b10/pkg/sql/logictest/logic.go#L2741)):*
<details><summary><i>Previously, yuzefovich wrote…</i></summary><blockquote>

Oh wait, we still don't check `resultMatches` here.
</blockquote></details>

ah i tried to be too smart


<!-- Sent from Reviewable.io -->

otan committed:
logictest: fix comparison logic

`9d2c054fd3e9e8c6a7d73f4e464ebd32ee82b8a2` introduced a bug in the
commit logic where it would only check logic test errors for the first
line, since it only compares the first column of results. This is now
fixed to compare every row.

Release note: None

yuzefovich reviewed (approved):
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55876)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

otan commented:
hurray nothing broke!

bors r=yuzefovich
